src/app/tests/TestTimedHandler.cpp

```diff
@@ -93,189 +115,153 @@ class TestExchangeDelegate : public Messaging::ExchangeDelegate
 
     delegate.mKeepExchangeOpen = true;
 
     CHIP_ERROR err = exchange->SendMessage(MsgType::TimedRequest, std::move(payload), SendMessageFlags::kExpectResponse);
-    NL_TEST_ASSERT(aSuite, err == CHIP_NO_ERROR);
+    EXPECT_EQ(err, CHIP_NO_ERROR);
 
-    ctx.DrainAndServiceIO();
-    NL_TEST_ASSERT(aSuite, delegate.mNewMessageReceived);
-    NL_TEST_ASSERT(aSuite, delegate.mLastMessageWasStatus);
-    NL_TEST_ASSERT(aSuite, delegate.mError == CHIP_NO_ERROR);
+    pTestContext->DrainAndServiceIO();
+    EXPECT_TRUE(delegate.mNewMessageReceived);
+    EXPECT_TRUE(delegate.mLastMessageWasStatus);
+    EXPECT_EQ(delegate.mError, CHIP_NO_ERROR);
 
     // Send an empty payload, which will error out but not with the
     // UNSUPPORTED_ACCESS status we expect if we miss our timeout.
     payload = MessagePacketBuffer::New(0);
-    NL_TEST_ASSERT(aSuite, !payload.IsNull());
+    ASSERT_FALSE(payload.IsNull());
 
     delegate.mKeepExchangeOpen   = false;
     delegate.mNewMessageReceived = false;
 
     err = exchange->SendMessage(aMsgType, std::move(payload), SendMessageFlags::kExpectResponse);
-    NL_TEST_ASSERT(aSuite, err == CHIP_NO_ERROR);
+    EXPECT_EQ(err, CHIP_NO_ERROR);
 
-    ctx.DrainAndServiceIO();
-    NL_TEST_ASSERT(aSuite, delegate.mNewMessageReceived);
-    NL_TEST_ASSERT(aSuite, delegate.mLastMessageWasStatus);
-    NL_TEST_ASSERT(aSuite, StatusIB(delegate.mError).mStatus != Status::UnsupportedAccess);
+    pTestContext->DrainAndServiceIO();
+    EXPECT_TRUE(delegate.mNewMessageReceived);
+    EXPECT_TRUE(delegate.mLastMessageWasStatus);
+    EXPECT_NE(StatusIB(delegate.mError).mStatus, Status::UnsupportedAccess);
 }
 
-void TestTimedHandler::TestInvokeFastEnough(nlTestSuite * aSuite, void * aContext)
+TEST_F(TestTimedHandler, TestInvokeFastEnough)
 {
-    TestFollowingMessageFastEnough(aSuite, aContext, MsgType::InvokeCommandRequest);
+    TestTimedHandler::TestFollowingMessageFastEnough(MsgType::InvokeCommandRequest);
 }
 
-void TestTimedHandler::TestWriteFastEnough(nlTestSuite * aSuite, void * aContext)
+TEST_F(TestTimedHandler, TestWriteFastEnough)
 {
-    TestFollowingMessageFastEnough(aSuite, aContext, MsgType::WriteRequest);
+    TestFollowingMessageFastEnough(MsgType::WriteRequest);
 }
 
-void TestTimedHandler::TestFollowingMessageTooSlow(nlTestSuite * aSuite, void * aContext, MsgType aMsgType)
+void TestTimedHandler::TestFollowingMessageTooSlow(MsgType aMsgType)
 {
-    TestContext & ctx = *static_cast<TestContext *>(aContext);
+    // TestContext & ctx = *static_cast<TestContext *>(aContext);
 
     System::PacketBufferHandle payload;
-    GenerateTimedRequest(aSuite, 50, payload);
+    GenerateTimedRequest(50, payload);
 
     TestExchangeDelegate delegate;
-    ExchangeContext * exchange = ctx.NewExchangeToAlice(&delegate);
-    NL_TEST_ASSERT(aSuite, exchange != nullptr);
+    ExchangeContext * exchange = pTestContext->NewExchangeToAlice(&delegate);
+    EXPECT_NE(exchange, nullptr);
```

Comment:
```suggestion
    ASSERT_NE(exchange, nullptr);
```
since exchange gets dereferenced after this